src/client/Microsoft.Identity.Client/Instance/CiamAuthority.cs

```diff
@@ -0,0 +1,20 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.Identity.Client.Instance
+{
+    internal class CiamAuthority : AadAuthority
+    {
+        internal CiamAuthority(AuthorityInfo authorityInfo) : 
+            base(authorityInfo)
+        {
+
```

Comment:
Not sure how to differentiate between CIAM and AAD authority yet. They seem to behave very similarly.  Need to investigate further.

Reply:
What do you mean by differentiate? from which point of view?

Reply:
FYI:

1. Currently I use "hostname ends with .ciamlogin.com" to differentiate CIAM from AAD.
2. They behave largely the same, except CIAM authority does not have instance discovery endpoint. So, I use `#1` to bypass `#2`.

Reply:
From the point of how MSAL should MSAL use the CIAM authority to generate the various endpoints (authorize, token, device code) as opposed to the AAD Authority. Right now the new `CiamAuthoirty` type is simple inheriting functionality from the `AddAuthority` and the tests seems to be working so I am not sure I need to add anything in the `CiamAuthoirty` class